pkg/controller/postgresqldatabase/postgresqldatabase_controller_test.go

```diff
@@ -0,0 +1,272 @@
+package postgresqldatabase
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+func TestReconcilePostgreSQLDatabase_ensurePostgreSQLDatabase_sunshine(t *testing.T) {
+	postgresqlHost := os.Getenv("POSTGRESQL_CONTROLLER_INTEGRATION_HOST")
+	if postgresqlHost == "" {
+		t.Skip("Integration test host not specified")
+	}
+	connectionString := fmt.Sprintf("postgresql://iam_creator:@%s?sslmode=disable", postgresqlHost)
+	db, err := postgresqlConnection(connectionString)
+	if err != nil {
+		t.Fatalf("connect to database failed: %v", err)
+	}
+	logger := testLogger{t: t}
+	logf.SetLogger(logf.ZapLoggerTo(&logger, true))
+
+	r := ReconcilePostgreSQLDatabase{
+		db: db,
+	}
+
+	name := fmt.Sprintf("test_%d", time.Now().UnixNano())
+	password := "test"
+
+	err = r.ensurePostgreSQLDatabase(logf.Log, name, password)
+	if err != nil {
+		t.Fatalf("ensurePostgreSQLDatabase failed: %v", err)
+	}
+
+	serviceConnectionString := fmt.Sprintf("postgresql://%s:%s@%s/%s?sslmode=disable", name, password, postgresqlHost, name)
+	db, err = postgresqlConnection(serviceConnectionString)
+	if err != nil {
+		t.Fatalf("connect to database failed: %v", err)
+	}
+
+	// Validate Schema
+	schemas := storedSchema(t, db, name)
+	assert.Equal(t, []string{name}, schemas, "schema not as expected")
+
+	// Validate iam_creator not able to see schema
+	schemas = storedSchema(t, r.db, name)
+	assert.Equal(t, []string(nil), schemas, "schema not as expected")
+
+	// Validate owner of database
+	owners := validateOwner(t, r.db, name)
+	t.Log(owners)
+	assert.Equal(t, []string{name}, owners, "owner not as expected")
+
+}
+
+func TestReconcilePostgreSQLDatabase_ensurePostgreSQLDatabase_idempotency(t *testing.T) {
+	postgresqlHost := os.Getenv("POSTGRESQL_CONTROLLER_INTEGRATION_HOST")
+	if postgresqlHost == "" {
+		t.Skip("Integration test host not specified")
+	}
+	connectionString := fmt.Sprintf("postgresql://iam_creator:@%s?sslmode=disable", postgresqlHost)
+	db, err := postgresqlConnection(connectionString)
+	if err != nil {
+		t.Fatalf("connect to database failed: %v", err)
+	}
+	logger := testLogger{t: t}
+	logf.SetLogger(logf.ZapLoggerTo(&logger, true))
+
+	r := ReconcilePostgreSQLDatabase{
+		db: db,
+	}
+
+	name := fmt.Sprintf("test_%d", time.Now().UnixNano())
+	password := "test"
+
+	err = r.ensurePostgreSQLDatabase(logf.Log, name, password)
+	if err != nil {
+		t.Fatalf("ensurePostgreSQLDatabase failed: %v", err)
+	}
+
+	// Invoke again with same name
+	err = r.ensurePostgreSQLDatabase(logf.Log, name, password)
+	if err != nil {
+		t.Logf("%#v", err)
+		t.Fatalf("ensurePostgreSQLDatabase failed: %v", err)
```

Comment:
Could be make this a bit more precise? Maybe:
```suggestion
		t.Fatalf("Second ensurePostgreSQLDatabase failed: %v", err)
```